h2o-core/src/main/java/water/util/Log.java

```diff
@@ -25,278 +27,284 @@
  **/
 abstract public class Log {
 
-  public static final byte FATAL= 0;
-  public static final byte ERRR = 1;
-  public static final byte WARN = 2;
-  public static final byte INFO = 3;
-  public static final byte DEBUG= 4;
-  public static final byte TRACE= 5;
-  public static final String[] LVLS = { "FATAL", "ERRR", "WARN", "INFO", "DEBUG", "TRACE" };
-  static String LOG_DIR = null;
-  static int _level=INFO;
-  static boolean _quiet = false;
-  private static org.apache.log4j.Logger _logger = null;
-  // Common pre-header
-  private static String _preHeader;
-  // A little bit of startup buffering
-  private static ArrayList<String> INIT_MSGS = new ArrayList<>();
-  
-  public static byte valueOf( String slvl ) {
-    if( slvl == null ) return -1;
-    slvl = slvl.toLowerCase();
-    if( slvl.startsWith("fatal") ) return FATAL;
-    if( slvl.startsWith("err"  ) ) return ERRR;
-    if( slvl.startsWith("warn" ) ) return WARN;
-    if( slvl.startsWith("info" ) ) return INFO;
-    if( slvl.startsWith("debug") ) return DEBUG;
-    if( slvl.startsWith("trace") ) return TRACE;
-    return -1;
-  }
-  
-  public static void init(String sLvl, boolean quiet) {
-    int lvl = valueOf(sLvl);
-    if( lvl != -1 ) _level = lvl;
-    _quiet = quiet;
-  }
-  
-  public static void setLogLevel(String sLvl, boolean quiet) {
-    init(sLvl, quiet);
-  }
-  
-  public static void setLogLevel(String sLvl) {
-    setLogLevel(sLvl, true);
-  }
-
-  public static void trace( Object... objs ) { log(TRACE,objs); }
-
-  public static void debug( Object... objs ) { log(DEBUG,objs); }
-  
-  public static void info ( Object... objs ) { log(INFO ,objs); }
-  
-  public static void warn ( Object... objs ) { log(WARN ,objs); }
-  
-  public static void err  ( Object... objs ) { log(ERRR ,objs); }
-  
+  public enum Level {
+    FATAL((byte)0, "fatal"),
+    ERROR((byte)1, "error"),
+    WARN((byte)2, "warn"),
+    INFO((byte)3, "info"),
+    DEBUG((byte)4, "debug"),
+    TRACE((byte)5, "trace");
+
+    private byte numLevel;
+    private String logFileName;
+    Level(byte numLevel, String logFileName) {
+      this.numLevel = numLevel;
+      this.logFileName = logFileName;
+    }
+
+    public String getLogFileName(){
+      return logFileName;
+    }
+
+    public byte getLevel(){
+      return numLevel;
+    }
+
+    public static Level fromString(String level) {
```

Comment:
This is a "slightly" breaking change, in the previous implementation people could specify "err" and it would correctly identify ERROR level.